cmd/tink-cli/cmd/delete/delete.go

```diff
@@ -0,0 +1,100 @@
+package delete
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/tinkerbell/tink/client"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type Options struct {
+	// DeleteByID is used to delete a resource
+	DeleteByID func(context.Context, *client.FullClient, string) (interface{}, error)
+
+	clientConnOpt *client.ConnOptions
+	fullClient    *client.FullClient
+}
+
+func (o *Options) SetClientConnOpt(co *client.ConnOptions) {
+	o.clientConnOpt = co
+}
+
+func (o *Options) SetFullClient(cl *client.FullClient) {
+	o.fullClient = cl
+}
+
+const shortDescr = "delete one or many resources"
+
+const longDescr = `Deletes one or many resources and prints the status of
+the deleted resource.`
+
+const exampleDescr = `# Delete template resource(s)
```

Comment:
Here you should add a reference to deleting more than one ID.

`tink hardware delete [id1] [id2][id3]`

You can place an example of the output as well.

As part of the long description you can add an explanation about the output format and how you can pipe it with other commands like awk.